Runtime/Scripts/BlendShapesController.cs

```diff
@@ -47,107 +52,130 @@ public class BlendShapesController : MonoBehaviour, IUsesStreamReader
         [Tooltip("Overrides settings for individual blend shapes.")]
         BlendShapeOverride[] m_Overrides;
 
-        readonly Dictionary<SkinnedMeshRenderer, BlendShapeIndexData[]> m_Indices = new Dictionary<SkinnedMeshRenderer, BlendShapeIndexData[]>();
-        float[] m_BlendShapes;
+        Dictionary<SkinnedMeshRenderer, BlendShapeIndexData[]> m_Indices = new Dictionary<SkinnedMeshRenderer, BlendShapeIndexData[]>();
+        Vector2[] m_SmoothedBlendShapes;
+        BlendShapeValues m_BlendShapeOutput;
 
-        IStreamSettings m_LastStreamSettings;
+        public BlendShapeValues blendShapeInput
+        {
+            get { return m_BlendShapeValues; }
+            set { m_BlendShapeValues = value; }
+        }
 
-        public SkinnedMeshRenderer[] skinnedMeshRenderers { get { return m_SkinnedMeshRenderers; }}
-        public Dictionary<SkinnedMeshRenderer, BlendShapeIndexData[]> blendShapeIndices { get { return m_Indices; } }
+        public BlendShapeValues blendShapeOutput
+        {
+            get { return m_BlendShapeOutput; }
+        }
 
-        public float[] blendShapesScaled { get; private set; }
+        public bool isTrackingActive
+        {
+            get { return m_TrackingActive; }
+            set { m_TrackingActive = value; }
+        }
+
+        public SkinnedMeshRenderer[] skinnedMeshRenderers
+        {
+            get { return m_SkinnedMeshRenderers; }
+        }
+
+        public Dictionary<SkinnedMeshRenderer, BlendShapeIndexData[]> blendShapeIndices
+        {
+            get { return m_Indices; }
+        }
 
         public IStreamReader streamReader { private get; set; }
 
-        void Start()
+#if UNITY_EDITOR
+        float m_LastTime;
+
+        void OnEnable()
         {
-            var streamSource = streamReader.streamSource;
-            if (streamSource == null)
-            {
-                Debug.LogError("Disabling BlendShapesController. No stream source set.", this);
-                enabled = false;
-                return;
-            }
+            //Temp fix
+            UnityEditor.AssemblyReloadEvents.afterAssemblyReload -= OnAssemblyReload;
+            UnityEditor.AssemblyReloadEvents.afterAssemblyReload += OnAssemblyReload;
```

Comment:
If this is a temp fix what's the long-term fix here?

Reply:
I would like to remove the indices dictionary and use the mappings asset actively instead. Something like:
void SetLocationValue(float value, BlendShapeLocation location, BlendShapeMappings mappings, GameObject root)
{
	var mapping = default(BlendShapeMapping);
	if (mappings.TryGetMapping(location, out mapping))
	{
	    foreach (var entry in mapping)
	    {
	        var path = entry.path;
	        var indices = entry.indices;
	        var renderer = root.Find(path).GetComponent<SkinnedMeshRenderer>(); //Do some caching here
	        renderer.SetBlendShapeValue(value, indices); //use extension method for that
	    }
	}
}

Reply:
Can this just go in a static constructor?

Reply:
Yes, good suggestion. Moved into a static constructor in latest commit.